tests/taskomatic_tests/conftest.py

```diff
@@ -22,3 +28,37 @@ def cleanup(request):
     def db_cleanup():
         restore_db(database_handler.pg_testing)
     request.addfinalizer(db_cleanup)
+
+
+@pytest.fixture(scope="module")
+def tornado_app():
+    """Return app instance"""
+    return taskomatic.APP
+
+
+@pytest.fixture(scope="module")
+def taskomatic_server(tornado_app):
+    loop = ioloop.IOLoop()
+    loop.make_current()
+    socket, port = testing.bind_unused_port()
+    server = HTTPServer(tornado_app)
+    server.add_socket(socket)
+
+    thread = threading.Thread(target=loop.start)
+    thread.start()
+
+    url = "http://localhost:%d" % port
+    yield url
+
+    loop.add_callback(server.stop)
+    loop.add_callback(server.close_all_connections)
+    loop.add_callback(loop.stop)
+
+    thread.join()
+
+
+@pytest.fixture()
+def http_client():
+    client = AsyncHTTPClient()
+    yield client
+    client.close()
```

Comment:
this can be simplified and you don't need to take care of event loop yourself, something like the following code should be enough
```python
@pytest.fixture(scope=module)
def taskomatic_server(event_loop, unused_tcp_port)
    server = HTTPServer(taskomatic.APP)
    server.listen(unused_tcp_port)
    yield f"http://localhost:{unused_tcp_port}"
    server.stop()


@pytest.fixture()
def http_client(taskomatic_server):
    client = AsyncHTTPClient()
    yield client
    client.close()

```
it would use asyncio event loop instead of the tornado event loop, but that should work
a test would be something like
```python
@pytest.mark.asyncio
async def test_something(http_client, taskomatic_server):
    resp = await http_client.fetch(f"{taskomatic_server}/api/something/else")
```

... or maybe even run a separate server for each test to isolate them (which might not be needed in this case)

just a suggestion off the top of my head, no need to implement it if the current solution works fine

Reply:
Yes, I know that this is possible. But I wanted to have the tornado loop in the test, because it is used in the taskomatic, this code I have made is fairly easy I think. :)

Reply:
sure, makes sense